server/build/webpack.js

```diff
@@ -156,6 +156,16 @@ export default async function createCompiler (dir, { dev = false, quiet = false
     loader: 'babel-loader',
     include: [dir],
     exclude (str) {
+      for (const pattern of config.transpileModules || []) {
+        if (!(pattern instanceof RegExp)) {
+          const message = `Incorrect pattern in config.transpileModules: "${pattern}".` +
+            'It accepts an array of regular expression'
+          throw new Error(message)
+        }
+
+        if (pattern.test(str)) return false
+      }
+
       return /node_modules/.test(str)
```

Comment:
would be cool if this was more generic to support monorepo scenarios. I'm happy to lend a hand to this feature if needed